src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -199,10 +200,9 @@ private IServiceCollection BuildHostingServices()
                 if (defaultPlatformServices.Application != null)
                 {
                     var appEnv = defaultPlatformServices.Application;
-                    if (!string.IsNullOrEmpty(_options.ApplicationBasePath))
-                    {
-                        appEnv = new WrappedApplicationEnvironment(_options.ApplicationBasePath, appEnv);
-                    }
+                    var applicationBasePath = ResolvePath(_options.ApplicationBasePath, appEnv.ApplicationBasePath);
+                    var environmentApplicationName = ResolveStartupAssemblyName() ?? appEnv.ApplicationName;
```

Comment:
I don't think there will ever be a case where we default to `appEnv.ApplicationName` since we need to specify a startup assembly/type/method to build the webhost. But I guess keeping the fallback doesn't hurt.